operator/flags.go

```diff
@@ -42,6 +42,12 @@ func init() {
 	flags.Float64(option.IPAMAPIQPSLimit, defaults.IPAMAPIQPSLimit, "Queries per second limit when accessing external IPAM APIs")
 	option.BindEnv(option.IPAMAPIQPSLimit)
 
+	flags.String(option.AzureSubscriptionID, "", "Subscription ID to access Azure API")
+	option.BindEnvWithLegacyEnvFallback(option.AzureSubscriptionID, "AZURE_SUBSCRIPTION_ID")
```

Comment:
Why are we binding `AzureSubscriptionID` with the legacy env `AZURE_SUBSCRIPTION_ID`? It's the first time that we are introducing azure so how can this flag be legacy?

Reply:
https://github.com/cilium/cilium/pull/10089#discussion_r385718811